edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/edgedeployment/service/KubernetesServiceMapper.cs

```diff
@@ -75,14 +75,21 @@ Option<V1Service> PrepareService(IModuleIdentity identity, KubernetesModule modu
                 //
                 // If the user wants to expose the ClusterIPs port externally, they should manually create a service to expose it.
                 // This gives the user more control as to how they want this to work.
-                var serviceType = onlyExposedPorts
-                    ? PortMapServiceType.ClusterIP
-                    : this.defaultMapServiceType;
+                var serviceType = this.GetServiceType(module, onlyExposedPorts);
+                var loadBalancerIP = module.Config.CreateOptions.ServiceOptions.FlatMap(so => so.LoadBalancerIP).OrDefault();
 
-                return Option.Some(new V1Service(metadata: serviceMeta, spec: new V1ServiceSpec(type: serviceType.ToString(), ports: servicePorts.Values.ToList(), selector: labels)));
+                return Option.Some(new V1Service(metadata: serviceMeta, spec: new V1ServiceSpec(type: serviceType.ToString(), loadBalancerIP: loadBalancerIP, ports: servicePorts.Values.ToList(), selector: labels)));
             }
 
             return Option.None<V1Service>();
         }
+
+        PortMapServiceType GetServiceType(KubernetesModule module, bool onlyExposedPorts)
```

Comment:
What's the thought process behind this? Why can't we just let k8s handle it?

Reply:
We have to choose a type for the service.  The translation logic in line 78 used to be:
                var serviceType = onlyExposedPorts
                    ? PortMapServiceType.ClusterIP
                    : this.defaultMapServiceType;

But with the addition of the service Options, that became more complicated to read, so I pulled this out into this subroutine. Now, the logic is, if the user specifies a service type, we use that. If the user does not, we use the old logic.

Reply:
I see. We are facing the same situation as the volume.